crates/typst-syntax/src/parser.rs

```diff
@@ -657,7 +655,12 @@ fn embedded_code_expr(p: &mut Parser) {
     p.enter_newline_mode(NewlineMode::Stop);
     p.enter(LexMode::Code);
     p.assert(SyntaxKind::Hash);
-    p.unskip();
+    if p.had_trivia() {
+        p.expected_at(p.before_trivia(), "expression");
+        p.exit();
+        p.exit_newline_mode();
+        return;
+    }
```

Comment:
why did this change?

Reply:
The previous code handled trivia following a hash (i.e. `#/**/var`) by just doing an `unskip` and expecting that the subsequent `code_expr_prec` call would create an error for an expected expression. In removing `unskip`  we now manually handle this case. I added a comment in the new commit